service/resolver/registry.go

```diff
@@ -80,7 +80,7 @@ func RegistryHostsFromConfig(cfg Config, credsFuncs ...Credential) source.Regist
 			tr := client.StandardClient()
 			if h.RequestTimeoutSec >= 0 {
 				if h.RequestTimeoutSec == 0 {
-					tr.Timeout = defaultRequestTimeoutSec * time.Second
+					tr.Timeout = time.Duration(defaultRequestTimeoutSec) * time.Second
```

Comment:
Based on the [Duration](https://pkg.go.dev/time#Duration) doc, it's a nanosecond, so this (`time.Duration(defaultRequestTimeoutSec)`) will change the meaning of 30 second to 30 nanosecond right?

Reply:
based on how [go produces those time const](https://cs.opensource.google/go/go/+/refs/tags/go1.20.1:src/time/time.go;l=631) this conversion is not needed. (int * time.Duration is more clear than time.Duration * time.Duration)

Reply:
See a few lines below for an example of this pattern.
Yes, `time.Duration(someSeconds)` does temporarily create a duration in nanoseconds, but the immediately following `* time.Second` corrects this.
There are other examples of this pattern at https://github.com/awslabs/soci-snapshotter/blob/f657479e6fdfae8a6a0516e6ba915e6e2dd2e807/fs/fs.go#L152-L162 and https://github.com/awslabs/soci-snapshotter/blob/f657479e6fdfae8a6a0516e6ba915e6e2dd2e807/fs/remote/resolver.go#L140-L141

Reply:
not aware of `int64 * time.Duration` is invalid... then time.Duration * time.Duration might be the only consistent way of doing that.

Also notice go itself uses both...: https://cs.opensource.google/go/go/+/refs/tags/go1.20.1:src/time/time.go;l=627-638